docs/function_guide.rst

```diff
@@ -0,0 +1,264 @@
+==================================
+Writing libEnsemble User Functions
+==================================
+
+libEnsemble coordinates ensembles of calculations performed by three main
+functions: a :ref:`Generator Function<api_gen_f>`, a :ref:`Simulator Function<api_sim_f>`,
+and an :ref:`Allocation Functions<api_alloc_f>`, or ``gen_f``, ``sim_f``, and
+``alloc_f`` respectively. These are all referred to as User Functions. Although
+libEnsemble includes several ready-to-use User Functions like
+:doc:`APOSMM<examples/aposmm>`, it's expected that most users will write their own.
```

Comment:
most users -> many users?
I think many are likely to reuse allocation functions, not sure what proportion of users will re-use gens.

Reply:
I would say something like "we expect most users will to be able to adjust existing user functions for their use cases."